x-pack/plugins/enterprise_search/public/applications/index.tsx

```diff
@@ -62,6 +64,10 @@ export const renderApp = (
     config,
     productAccess,
     charts: plugins.charts,
+    data: plugins.data,
+    lens: plugins.lens,
+    defaultDataView,
```

Comment:
Trying to understand where the defaultDataView should sit. I would like to steer away from having it within the application context, though right now I dont have a better suggestion. Ideally it would be only for engines + analytics collections views.

Reply:
Moved fetching `defaultDataView ` and `formula` to engine_analytics_logic.ts [listeners](https://github.com/elastic/kibana/blob/8364a5088dbfcdf2133dddfad493c306571e4680/x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/engine/engine_analytics_logic.ts#L94). This can prevent from making [index.tsx](https://github.com/elastic/kibana/blob/8364a5088dbfcdf2133dddfad493c306571e4680/x-pack/plugins/enterprise_search/public/applications/index.tsx#L37)  `async`

Reply:
For the analytics collections view - I think we may not need the defaultDataView only `formula` which could be consumed via this logic file or have some common index file for both the components? (not sure of this approach though)

Reply:
Also may be move `Formula` to [shared](https://github.com/elastic/kibana/tree/18502f18e1e7e4fad3e539cdd336a05b3c8b2bf2/x-pack/plugins/enterprise_search/public/applications/shared) folder
Path: `/x-pack/plugins/enterprise_search/public/applications/shared `